ts/handler_utils/base.py

```diff
@@ -0,0 +1,411 @@
+import logging
+import os
+import time
+
+import psutil
+import torch
+
+from ts.handler_utils.caller import InitCaller, PipeCaller
+from ts.utils.util import (
+    list_classes_from_module,
+    load_compiler_config,
+    load_label_mapping,
+)
+
+logger = logging.getLogger(__name__)
+# Possible values for backend in utils.py
+def check_pt2_enabled():
+    try:
+        import torch._dynamo
+
+        pt2_enabled = True
+        if torch.cuda.is_available():
+            # If Ampere enable tensor cores which will give better performance
+            # Ideally get yourself an A10G or A100 for optimal performance
+            if torch.cuda.get_device_capability() >= (8, 0):
+                torch.backends.cuda.matmul.allow_tf32 = True
+    except ImportError as error:
+        logger.warning(
+            "dynamo/inductor are not installed. \n For GPU please run pip3 install numpy --pre torch[dynamo] --force-reinstall --extra-index-url https://download.pytorch.org/whl/nightly/cu117 \n for CPU please run pip3 install --pre torch --extra-index-url https://download.pytorch.org/whl/nightly/cpu"
+        )
+        pt2_enabled = False
+    return pt2_enabled
+
+
+def _load_torchscript_model(obj, model_pt_path):
+    """Loads the PyTorch model and returns the NN model object.
+    Args:
+        model_pt_path (str): denotes the path of the model file.
+    Returns:
+        (NN Model Object) : Loads the model object.
+    """
+    return torch.jit.load(model_pt_path, map_location=obj.device)
+
+
+def _load_pickled_model(obj, model_dir, model_file, model_pt_path):
+    """
+    Loads the pickle file from the given model path.
+    Args:
+        model_dir (str): Points to the location of the model artefacts.
+        model_file (.py): the file which contains the model class.
+        model_pt_path (str): points to the location of the model pickle file.
+    Raises:
+        RuntimeError: It raises this error when the model.py file is missing.
+        ValueError: Raises value error when there is more than one class in the label,
+                    since the mapping supports only one label per class.
+    Returns:
+        serialized model file: Returns the pickled pytorch model file
+    """
+    model_def_path = os.path.join(model_dir, model_file)
+    if not os.path.isfile(model_def_path):
+        raise RuntimeError("Missing the model.py file")
+
+    module = importlib.import_module(model_file.split(".")[0])
+    model_class_definitions = list_classes_from_module(module)
+    if len(model_class_definitions) != 1:
+        raise ValueError(
+            "Expected only one class as model definition. {}".format(
+                model_class_definitions
+            )
+        )
+
+    model_class = model_class_definitions[0]
+    model = model_class()
+    if model_pt_path:
+        state_dict = torch.load(model_pt_path, map_location=obj.device)
+        model.load_state_dict(state_dict)
+    return model
+
+
+def base_initialize(obj, context):
```

Comment:
you were probably already thinking this but worth splitting this out even more the IPEX, TensorRT, PT 2 logic shouldnt be in the base handler proper but in its own file maybe compiler_optimizations.py or something - regardless i was planning on taking a crack at this part so we can worry about it later